calc_from_args_utils/udf_arg_methods.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+import logging
+from typing import Any
+
+import yaml
+from genologics.entities import Artifact, Process
+
+from epp_utils import udf_tools
+
+
+def fetch_from_arg(
+    art_tuple: tuple, arg_dict: dict, process: Process, on_fail=AssertionError
+) -> Any:
+    """Branching decision-making function. Determine HOW to fetch UDFs given the argument dictionary."""
```

Comment:
Same here, I would like to be able to read which keys/values from arg_dict that are required and which are used. But maybe it's too complicated?

Reply:
d912aa8c7112cd1d69020d77943912b6c91082e7